### PR TITLE
[ISSUE #125]Correct flaky tests for PUT articles endpoint

### PR DESCRIPTION
**Pull request by Justyna-KO**
### Scope of changes:

I changed the range of randomNumber used as the id of the articles.

### How to use it:

..
## Any other comments?

...

Before submitting Pull Request, I've ensured that:

- [ ] The PR is associated with the relevant Issue before its creation.
- [ ] All new and existing tests passed
- [ ] My commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by Slawomir-DKl — CHANGES_REQUESTED** [review]

File: tests/api/articles/article-id-put.spec.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { customDate } from 'test-data/shared/date.generator';
 
 test.describe('PUT articles/{id} endpoint tests', async () => {
   const articles = `/api/articles`;
-  const randomNumber = Math.floor(Math.random() * (1000 - 100 + 1)) + 100;
+  const randomNumber = Math.floor(Math.random() * (2000 - 1001 + 1)) + 1001;
```
[the remark this answers is not in the document]
Comment (Justyna-KO) [reply tests/api/articles/article-id-put.spec.ts:10]: I added articleID.generator.ts.

**Review by Slawomir-DKl — COMMENTED** [review]
Great changes. I've learned a lot from your solutions and now I'll introduce some changes to my PR to be consistent with your ideas such as map() instead of for loop (which I find much better)

I have only two little remarks - for your consideration

File: test-data/shared/articleID.generator.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { APIRequestContext, APIResponse } from '@playwright/test';
+
+// generate unique article ID
+export async function generateUniqueArticleId(
+  request: APIRequestContext,
+  minRange: number = 1001,
+  maxRange: number = 2000,
+): Promise<number> {
+ // get list of all article IDs
```
[the remark this answers is not in the document]
Comment (Justyna-KO) [reply test-data/shared/articleID.generator.ts:9]: ok

File: test-data/shared/articleID.generator.ts
[the remark this answers is not in the document]
Comment (Justyna-KO) [reply test-data/shared/articleID.generator.ts:1]: I agree.

File: tests/api/articles/article-id-put.spec.ts [hunk 3]
```diff
@@ -2,11 +2,12 @@ import { HttpStatusCode } from '@_src_api/enums/api-status-code.enum';
 import { testUsers } from '@_src_fixtures_api/auth';
 import { createHeaders } from '@_src_helpers_api/create-token.helper';
 import { APIResponse, expect, test } from '@playwright/test';
+import { generateUniqueArticleId } from 'test-data/shared/articleID.generator';
```
Comment (Slawomir-DKl) [inline tests/api/articles/article-id-put.spec.ts:5]: Thank you for the changes. After changing the generator file name, you have to change the module to article.generator as well

**Review by Slawomir-DKl — CHANGES_REQUESTED** [review]

**Review by Slawomir-DKl — APPROVED** [review]
Thank you again :)

**Pull request merged**